OpenRA.Mods.Common/Traits/Cargo.cs

```diff
@@ -214,7 +214,7 @@ public bool CanUnload()
 			}
 
 			return !IsEmpty(self) && (aircraft == null || aircraft.CanLand(self.Location))
-				&& CurrentAdjacentCells != null && CurrentAdjacentCells.Any(c => Passengers.Any(p => p.Trait<IPositionable>().CanEnterCell(c)));
+				&& CurrentAdjacentCells != null && CurrentAdjacentCells.Any(c => Passengers.Any(p => p.Trait<IPositionable>().CanEnterCell(c, null, false)));
```

Comment:
This will return the wrong result for cases like `EjectOnDeath: true` that need to know if it can unload an actor right now, which will lead to units being stacked on top of eachother in the same (sub)cell.

Reply:
Done.

Reply:
It turns out that pretty much everything else in the `EjectOnDeath` code is bogus, and so this issue already exists on bleed, and still exists after the changes here - see #16353 for details.

I can confirm that the change here does work after hacking around the other issues, which we shouldn't be trying to fix in this PR.